### DISCUSSION TITLE
Intuitive database searching

### DISCUSSION THREAD

**Discussion by eboileau** [dieterich-lab/scimodom#68] [Features]
### Background
We'd prefer not to restrict too much the search, but on the other hand, sending back thousands of records across different dataset (different modifications, techniques, species, *etc*) may be too much information anyway.

### What we currently do
Currently, we allow in principle to select, among available data, *(1)* one, more, or all modifications (or RNA type), *(2)* one, more, or all techniques, and *(3)* one, more, or all organisms, in that order. See [Search](https://scimodom-beta.dieterichlab.org/). 

**Pros:** You can find site-specific modifications across organisms, techniques, w/o restrictions.
**Cons:** Large data, longer query time, not sustainable for the future in that format

### What others do
Other databases allow to first select species, then cell or tissue, but display data for one organism at a time, and do not allow to filter based on the technique used (*e.g.* eTAM-seq, GLORI, *etc*).

### Your feedback
**How do you want to search the database? What is intuitive for you?**

#### Suggested answers:
- It's important to display data across organisms, not just one organism at a time.
- I'm only interested in one organism at a time.
- I like to search by technique, then by available organism (cell, tissue, organ).

**Comment by inaarmann** [comment]
Hi @eboileau ,

for me the relevance in the "Search" would be:
1. RNA modification
2. RNA biotype (mRNA, rRNA, tRNA)
3. organism
4. technology

1 & 2 are currently in one field, but if e.g. tRNAs are included, one could think of separating these.

I assume that in the "Search" field, most people will look for RNA modifications in one specific organism, whereas more than one organism may be relevant for "Compare" (e.g. shared m6A sites between mouse and human).

It would be nice if it would be possible to filter further via the column headers (e.g. for a specific gene).
Also, a link to the study from which the data are derived would be great.

Possible Bug: I tested the "Export" function in the Search, and it seems that currently only the displayed page is exported.

**Reply by eboileau** [reply]
Thanks @inaarmann for your feedback. Table filtering is coming. As for export, it is intended, since only what is shown can be downloaded. More data download options will eventually be available.

**Reply by eboileau** [reply]
> I assume that in the "Search" field, most people will look for RNA modifications in one specific organism, whereas more than one organism may be relevant for "Compare" (e.g. shared m6A sites between mouse and human).

In the Compare view, the comparisons are based *e.g.* on intersection of intervals, so must share same chromosome nomenclature, *etc.* The results can be unexpected if we have different chromosome lengths, and difficult to interpret for the user. 

If we want to be able to search for sites across different species, then we need to allow this in Search, possibly revisiting the filtering options, or else we have to create another view/tab for cross-organism search.
